measures/Measures/MCFrame.cc

```diff
@@ -466,19 +468,22 @@ void MCFrame::makePosition(const MeasFrame& frame) {
   impl_->posGeop = MVPosition();
 }
 
-void MCFrame::makeDirection(const MeasFrame& frame) {
+void MCFrame::makeDirection(MeasFrame& frame) {
   static const MDirection::Ref REFJ2000 = MDirection::Ref(MDirection::J2000);
+  details::CyclicState state = frame.rep.Freeze();
   impl_->dirConvJ2000 = MDirection::Convert(*frame.direction(),
-					 MDirection::Ref(MDirection::J2000,
-							 frame));
+          MDirection::Ref(MDirection::J2000, frame));
+  frame.rep.Unfreeze(state);
 
   static const MDirection::Ref REFB1950 = MDirection::Ref(MDirection::B1950);
+  state = frame.rep.Freeze();
```

Comment:
Can't you remove these new statements, so only one Freeze..UnFreeze block remains?
```suggestion

  static const MDirection::Ref REFB1950 = MDirection::Ref(MDirection::B1950);
```